airbyte-integrations/connectors/source-intercom/source_intercom/source.py

```diff
@@ -0,0 +1,324 @@
+    Endpoint: https://api.intercom.io/admins
+    """
+
+    primary_key = "id"
+    data_fields = ["admins"]
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "admins"
+
+
+class Companies(IncrementalIntercomStream):
+    """Return list of all companies.
+    API Docs: https://developers.intercom.com/intercom-api-reference/reference#list-companies
+    Endpoint: https://api.intercom.io/companies
+    """
+
+    primary_key = "id"
+    data_fields = ["data"]
+    cursor_field = "updated_at"
+
+    def path(self, **kwargs) -> str:
+        return "companies"
+
+
+class CompanySegments(StreamMixin, IncrementalIntercomStream):
+    """Return list of all company segments.
+    API Docs: https://developers.intercom.com/intercom-api-reference/reference#list-attached-segments-1
+    Endpoint: https://api.intercom.io/companies/<id>/segments
+    """
+
+    primary_key = "id"
+    data_fields = ["data"]
+    cursor_field = "updated_at"
+    stream = Companies
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return f"/companies/{stream_slice['id']}/segments"
+
+
+class Conversations(IncrementalIntercomStream):
+    """Return list of all conversations.
+    API Docs: https://developers.intercom.com/intercom-api-reference/reference#list-conversations
+    Endpoint: https://api.intercom.io/conversations
+    """
+
+    primary_key = "id"
+    data_fields = ["conversations"]
+    cursor_field = "updated_at"
+
+    def path(self, **kwargs) -> str:
+        return "conversations"
+
+
+class ConversationParts(StreamMixin, IncrementalIntercomStream):
+    """Return list of all conversation parts.
+    API Docs: https://developers.intercom.com/intercom-api-reference/reference#retrieve-a-conversation
+    Endpoint: https://api.intercom.io/conversations/<id>
+    """
+
+    primary_key = "id"
+    data_fields = ["conversation_parts", "conversation_parts"]
+    cursor_field = "updated_at"
+    stream = Conversations
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return f"/conversations/{stream_slice['id']}"
+
+
+class Segments(IncrementalIntercomStream):
+    """Return list of all segments.
+    API Docs: https://developers.intercom.com/intercom-api-reference/reference#list-segments
+    Endpoint: https://api.intercom.io/segments
+    """
+
+    primary_key = "id"
+    data_fields = ["segments"]
+    cursor_field = "updated_at"
+
+    def path(self, **kwargs) -> str:
+        return "segments"
+
+
+class Contacts(IncrementalIntercomStream):
+    """Return list of all contacts.
+    API Docs: https://developers.intercom.com/intercom-api-reference/reference#list-contacts
+    Endpoint: https://api.intercom.io/contacts
+    """
+
+    primary_key = "id"
+    data_fields = ["data"]
```

Comment:
If most of the streams are using `data_fields = ["data"]` make this a part of `IncrementalIntercomStream` class and override this property in the particular class as the stream need this change.